Changes for pALPIDE-3

mkofarag opened the pull request:
@freidt @hyeonjoongkim

hyeonjoongkim committed:
Included Backward compatibility for pALPIDE-2
	modified:   include/EUTelProcessorAnalysisPALPIDEfs.h
	modified:   jobsub/examples/pALPIDEfs/config_pALPIDEfs_7.cfg
	modified:   jobsub/examples/pALPIDEfs/steering-templates_pALPIDEfs_7/analysis-tmp.xml
	modified:   src/EUTelProcessorAnalysisPALPIDEfs.cc

hyeonjoongkim committed:
Added Version Check in pALPIDEfs Analysis
	modified:   jobsub/examples/pALPIDEfs/config_pALPIDEfs_7.cfg
	modified:   jobsub/examples/pALPIDEfs/runlist_example.csv
	modified:   jobsub/examples/pALPIDEfs/steering-templates_pALPIDEfs_7/analysis-tmp.xml
	modified:   src/EUTelProcessorAnalysisPALPIDEfs.cc

hyeonjoongkim committed:
Changed residual cuts for pALPIDE3
	modified:   EUTelProcessorAnalysisPALPIDEfs.cc

hyeonjoongkim committed:
Analysis to include chip Version properly
	modified:   src/EUTelProcessorAnalysisPALPIDEfs.cc

hyeonjoongkim committed:
Fake hit histograms including 8sectors
	modified:   EUTelProcessorAnalysisPALPIDEfs.cc

hyeonjoongkim committed:
Output of setting_DUT file changed
	modified:   ../jobsub/examples/pALPIDEfs/runlist_example.csv
	modified:   EUTelProcessorAnalysisPALPIDEfs.cc

hyeonjoongkim committed:
Discard HotPixelFinder and changed to NoisyPixelFinder
	modified:   hotpixel-tmp.xml

hyeonjoongkim committed:
AddressToColumn function modification
	modified:   EUTelProcessorAnalysisPALPIDEfs.cc

hyeonjoongkim committed:
Merge branch 'master' into mine

hyeonjoongkim committed:
changed declaration LeftRight
	modified:   EUTelProcessorAnalysisPALPIDEfs.cc

hyeonjoongkim committed:
xml file for hotpixel modified to give proper conf
	modified:   hotpixel-tmp.xml

hyeonjoongkim committed:
Merge branch 'master' into mine

	modified:   include/EUTelProcessorAnalysisPALPIDEfs.h
	modified:   jobsub/examples/pALPIDEfs/config_pALPIDEfs_7.cfg
	modified:   jobsub/examples/pALPIDEfs/steering-templates_pALPIDEfs_7/analysis-tmp.xml
	modified:   src/EUTelMille.cc
	modified:   src/EUTelProcessorAnalysisPALPIDEfs.cc

hyeonjoongkim committed:
Added backward compatibility for Monika's change
	modified:   EUTelProcessorAnalysisPALPIDEfs.cc

mkofarag commented on src/EUTelProcessorAnalysisPALPIDEfs.cc:
```diff
@@ -744,7 +763,7 @@ void EUTelProcessorAnalysisPALPIDEfs::processEvent(LCEvent *evt)
                 for (unsigned int i=0; i<chi2Max.size(); i++)
                 {
-                  if (chi2 < chi2Max[i] && yposfit < 12.2 && yposfit > 9.7 && xposfit < 26.2 && xposfit > 2.5)
+                  if (chi2 < chi2Max[i] && yposfit < 12.2 && yposfit > 9.7 && xposfit < 26.2+1.3 && xposfit > 2.5 -1.3)
```
Why is the extra 1.3?

mkofarag commented on src/EUTelProcessorAnalysisPALPIDEfs.cc:
```diff
@@ -762,7 +781,7 @@ void EUTelProcessorAnalysisPALPIDEfs::processEvent(LCEvent *evt)
                     nResidualYPixel2by2[chi2Max[i]][index]->Fill(fmod(xposfit,2*xPitch),fmod(yposfit,2*yPitch));
                   }
-                  else if (chi2 < chi2Max[i] && (yposfit < 9.2 || yposfit > 12.7 || xposfit > 27.4 || xposfit < 1.2))
+                  else if (chi2 < chi2Max[i] && (yposfit < 9.2 || yposfit > 12.7 || xposfit > 27.4+1.3 || xposfit < 1.2 - 0.2))
```
Why is the extra 1.3 and 0.2?

mkofarag replied:
I think these numbers, including the numbers to cut on the hole of the tracking planes should be parameters read from the steering templates.

hyeonjoongkim replied:
@mkofarag 
The cut hole parameters, I'll update those!
The funny thing is : measured y width with caliper is 5mm wide, but what we see in the analysis result is ~3mm wide.

mkofarag replied:
I think what we see in the analysis results is a convolution of the actual size of the hole with the misalignment and with the size of the hole in the other planes. The values in the code are a conservative estimate which I got by looking at the analysis results, not by measuring the actual size of the hole. I would be fine with updating these values with the actual measured ones, although they still should be variables being read from the steering templates, since different carriers have different hole sizes, so these values could change from measurement to measurement.

mkofarag commented on jobsub/examples/pALPIDEfs/steering-templates_pALPIDEfs_7/hotpixel-tmp.xml:
```diff
@@ -26,7 +26,7 @@
       <parameter name="LCIOInputFiles"> @LcioPath@/@FilePrefix@-converter.slcio </parameter>
       <parameter name="GearXMLFile" value="@GearFilePath@/@GearFile@"/>
-      <parameter name="MaxRecordNumber" value="@nEventsForHotpixel@"/>
+      <parameter name="MaxRecordNumber" value="@MaxRecordNumber@"/>
```
Where do you set MaxRecordNumber? I think this value could go into the config file.

hyeonjoongkim replied:
@mkofarag 
MaxRecordNumber is set inside the config file. 
I've changed those because there was some buggy stuff on-going when I set MaxRecordNumber = NoOfEvents

mkofarag replied:
Yes, I saw that also. Sorry, I thought MaxRecordNumber is a new variable and that's why I got confused where it is set. I see now it used to be there in the config file before the changes as well.

mkofarag commented:
I think this branch doesn't contain the changes to take into account only the tracks within the hole for the tracking planes as well. Could you merge that also into the same branch? @hyeonjoongkim

mkofarag commented on src/EUTelProcessorAnalysisPALPIDEfs.cc:
```diff
@@ -70,16 +70,17 @@ EUTelProcessorAnalysisPALPIDEfs::EUTelProcessorAnalysisPALPIDEfs()
+  nTracks(8),
+  nTracksPAlpide(8),
+  nTracksFake(8),
+  nTracksPAlpide(8),
```
This line is there twice.

hyeonjoongkim replied:
@mkofarag You mean nFake and nTracksFake is same value?
nTracksFake series are the one that is new feature introduced by you and your colleague,
and nFake was the one from original version.

mkofarag replied:
No, I mean  nTracksPAlpide(8) is there in line 74 and in line 76.

hyeonjoongkim committed:
Size of hole in Carrier board moved to steering templates
	modified:   jobsub/examples/pALPIDEfs/steering-templates_pALPIDEfs_7/analysis-tmp.xml
	modified:   src/EUTelProcessorAnalysisPALPIDEfs.cc

mkofarag commented:
Thanks for the change of moving the hole sizes to the steering template. I think the only thing missing now is the code you wrote to check whether a track is within the holes of the tracking planes as well.

hyeonjoongkim committed:
starting to work with c++03 compatiblity
	modified:   EUTelProcessorAnalysisPALPIDEfs.cc

Commit:
Fix some compilation errors

hyeonjoongkim committed:
Merge pull request #1 from lbar314/mine

Fix some compilation errors

mkofarag commented:
The issues I see open are the following:
- Cut on tracks within hole of tracking planes
- I see errors like this at the beginning of the clustering step. We need to check whether this is harmless or whether there is a problem with the new version of the hotpixel treatment:
  15:15:09 jobsub.clustering(ERROR): [ ERROR5 "Clustering_pALPIDEfs"] hot pixel [index 197439] reoccured ?!
  I see these only at the first time clustering is called
- Warnings in the code of the analysis step when compiling.

mkofarag commented:
I would also put
parameter name="MaxRecordNumber" value="20000"
instead of 
parameter name="MaxRecordNumber" value="@MaxRecordNumber@"
into hotpixel-tmp.xml. The reason 10000 did not work is that it was required to study 10000 good events, but the number of checked events were also limited to 10000. So if a few events were discarded from the first 10000 a problem occurred, because it could not check 10000 good events. I think we are safe with 20000 and this would speed up the hotpixel step.

hyeonjoongkim commented:
@mkofarag 
For MaxRecordNumber parameter, isn't it the limitations of nEventsForHotpixel this value?
if the processor reaches nEventsForHotpixel, it just skips all afterward events.
I've just put MaxRecordNumber because it doesn't affect speed so much..

mkofarag commented:
It doesn't effect the speed a lot, but since it still prints the runnumber every 1000 event if we have many events it actually causes a difference. It also produces unnecessarily large log files.

hyeonjoongkim commented:
@mkofarag 
Okay! I'll change those also

hyeonjoongkim committed:
compiling issue changed & hot-pixel maxrecordnumber changed
	modified:   jobsub/examples/pALPIDEfs/steering-templates_pALPIDEfs_7/hotpixel-tmp.xml
	modified:   src/EUTelProcessorAnalysisPALPIDEfs.cc

hyeonjoongkim committed:
Merge branch 'mine' of https://github.com/hyeonjoongkim/eutelescope into mine

Commit:
Removed warning messages
	modified:   include/EUTelProcessorAnalysisPALPIDEfs.h
	modified:   src/EUTelProcessorDeadColumnFinder.cc

hyeonjoongkim committed:
cleaning not-used processors
	modified:   hotpixel-tmp.xml

tbisanz committed:
Merge pull request #274 from mkofarag/master

Problem with noisy pixel collection

mkofarag commented:
Hi,

The pALPIDEfsEUTelAutoPedestalNoiseProcessor is not needed in the hotpixel steering template either, you could remove it completely.

hyeonjoongkim committed:
Merge branch 'master' into mine

hyeonjoongkim committed:
cleaning not-used processors
	modified:   hotpixel-tmp.xml
